src/kernels/api/api.ts

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { CancellationToken, Event, EventEmitter, Uri, workspace } from 'vscode';
+import { ExecutionResult, Kernel, Kernels } from '../../api';
+import { ServiceContainer } from '../../platform/ioc/container';
+import { IKernel, IKernelProvider } from '../types';
+import { executeSilentlyAndEmitOutput } from '../helpers';
+import { IDisposable } from '../../platform/common/types';
+import { dispose } from '../../platform/common/utils/lifecycle';
+import { noop } from '../../platform/common/utils/misc';
+
+const kernelCache = new WeakMap<IKernel, Kernel>();
+
+class WrappedKernel implements Kernel {
+    get status(): 'unknown' | 'starting' | 'idle' | 'busy' | 'terminating' | 'restarting' | 'autorestarting' | 'dead' {
+        return this.kernel.status;
+    }
+    get onDidChangeStatus(): Event<
+        'unknown' | 'starting' | 'idle' | 'busy' | 'terminating' | 'restarting' | 'autorestarting' | 'dead'
+    > {
+        return this.kernel.onStatusChanged;
+    }
+
+    constructor(private readonly kernel: IKernel) {}
+
+    executeCode(code: string, token: CancellationToken): ExecutionResult {
```

Comment:
Do we need to whitelist access to this?

Reply:
That will come later, for now this API is not exposed  (i.e. code is not implemented to return this API from the Jupyter extension)